pkg/apis/core/v1/core.proto

```diff
@@ -20,6 +20,7 @@ message BootstrapTokenMetadata {
   map<string, string> labels = 4;
   repeated TokenCapability capabilities = 5;
   string resourceVersion = 6;
+  bool onetime = 7;
```

Comment:
This should be implemented via a `maxUsages` field. Since usage counts are already tracked, that would integrate nicely with it.